packages/lesswrong/components/posts/PostsPage/PostsPageWrapper.tsx

```diff
@@ -57,15 +66,16 @@ const PostsPageWrapper = ({ sequenceId, version, documentId }: {
     } else {
       throw new Error(error.message);
     }
-  } else if (!post) {
+  } else if (!post && !postPreload) {
     return <Error404/>
-  } else if (isPostWithForeignId(post)) {
+  } else if (post && isPostWithForeignId(post)) {
     return <PostsPageCrosspostWrapper post={post} eagerPostComments={eagerPostComments} refetch={refetch} fetchProps={fetchProps} />
```

Comment:
`postPreload` should have `fmCrosspost` data in there, so if we don't have `post` we should be able to pass in `postPreload` instead right? That might help us avoid the situation where the crosspost first displays with no body text, then turns into a loading component, then finally displays the proper full post.